### PR TITLE
entities: some improve navigation and links

### PR DESCRIPTION
**Pull request by Garfield-fr**
* Adds a new tab for related entities on document detail view.
* Updates contribution, subject and genreForm to relaunch a search.
* Updates search documents on entities detail view to target certain fields only.
* rero/rero-ils#3513.

## Display
document-entities

## Dependencies
- https://github.com/rero/rero-ils/pull/3537

### CODE REVIEW

File: projects/admin/src/app/record/detail-view/document-detail-view/entities-related/entities-related.component.html [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+-->
+<div class="container">
+  <div class="row" *ngFor="let field of entities | keyvalue; let i = index">
+    <div [class]="i%2 === 0 ? 'bg-light':''" class="font-weight-bold col-3 py-1" translate>
```
Comment (zannkukai) [inline projects/admin/src/app/record/detail-view/document-detail-view/entities-related/entities-related.component.html:19]: `[class.bd-light]="!!(i%2)"` is prettier.
This class should be added on parent `div.row` element. All children div's will be colored

File: projects/shared/src/lib/view/entity-link.component.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        simple: '0',
+      };
+    } else {
+      this.queryParams = {
+        q: `${this.resourceName}.entity.authorized_access_point_${lang}:"${this.linkName}"`,
+        simple: '0',
+      }
+    }
+    if (this.external) {
+      this.externalHrefLink = Entity.generateHrefLink(this.routerLinkParams, this.queryParams);
```
Comment (zannkukai) [inline projects/shared/src/lib/view/entity-link.component.ts:89]: I'm not fan about "generatedHrefLink" method : It smells like "[routerLink]" and I don't understand why it's not used.

**Review by lauren-d — APPROVED** [review]

File: projects/admin/src/app/record/detail-view/document-detail-view/entities-related/entities-related.component.html [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+-->
+<div class="container">
+  <div [class]="i%2 === 0 ? 'bg-light':''" class="row" *ngFor="let field of entities | keyvalue; let i = index">
+    <div class="font-weight-bold col-3 py-1" translate>
```
Comment (jma) [inline projects/admin/src/app/record/detail-view/document-detail-view/entities-related/entities-related.component.html:19]: `translate` seems called twice.

File: projects/shared/src/lib/class/entity.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+  }
+
+  /**
+   * Generate the advanced search query
+   * @param resourceType - type of entity
+   * @param catalogKey - Catalog key
+   * @param catalogPid - Catalog pid
+   * @returns the search string
+   */
+  static generateEntitySearchQuery(resourceType: string, catalogKey: string, catalogPid: string): string {
```
Comment (jma) [inline projects/shared/src/lib/class/entity.ts:104]: I would rename this into `generateSearchQuery`

File: projects/shared/src/lib/view/entity-link.component.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    >{{ linkName }}</a>
+    </ng-template>
+  `
+})
+export class EntityLinkComponent implements OnInit {
+
+  /** Entity field metadata */
+  @Input() entity: any;
+
+  /** Entity field name */
```
Comment (jma) [inline projects/shared/src/lib/view/entity-link.component.ts:43]: field name or resource name?

File: projects/shared/src/lib/view/entity-link.component.ts [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  @Input() resourceName: string;
+
+  /** Link class */
+  @Input() class: string;
```
Comment (jma) [inline projects/shared/src/lib/view/entity-link.component.ts:47]: Can be confusing with the reserved word `class` you can use maybe `className`?

**Review by jma — APPROVED** [review]

**Review by zannkukai — APPROVED** [review]

**Pull request merged**